gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,40 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    @property
+    def pivot_energy(self):
+        """The decorrelation energy for a given spectral model calculated numerically.
+
+        Returns
+        -------
+        pivot energy : `~astropy.units.Quantity`
+            The energy at which the statistical error is smallest.
+            If no minimum is found, NaN will be returned.
+        """
+
+        x_unit = self.reference.unit
+
+        def min_func(x):
+            """Function to minimize"""
+            x = np.exp(x)
+            dnde, dnde_error = self.evaluate_error(x * x_unit)
+            return dnde_error / dnde
+
+        bounds = [np.log(self.reference.value) - 3, np.log(self.reference.value) + 3]
+
+        std = np.std(min_func(x=np.linspace(bounds[0], bounds[1], 100)))
+        if std < 1e-5:
+            log.warning("No minimum was found.")
+            return np.nan * x_unit
```

Comment:
do nans need units?

Reply:
What is this warning case?

Reply:
This is super important! minimize_scalar fails when the function is constant...

Reply:
ok

Reply:
> do nans need units?

Keeping units to be consistent with the analytical output for the `PowerLawSpectralModel`